pdata/README.md

```diff
@@ -0,0 +1,214 @@
+- `Sort(func(a, b <element type>) bool) <slice type>`: sorts the slice using the function and returns itself.
+
+### Repeated scalar fields
+
+Repeated scalar protobuf fields MUST be represented as another pdata type wrapping native go slices. Name of the 
+type MUST include name of the primitive data type ending with `Slice`, e.g. `UInt64Slice`. Each pdata type MUST have 
+the following methods:
+
+- `AsRaw() []<primitive type>`: returns copy of the underlying go slice. 
+- `FromRaw(val []<primitive type>)`: copies a go slice into the pdata wrapper instance.
+- `Len()`: returns length of the slice.
+- `At(int) <primitive type>`: returns an element from the slice by its index, can cause "index out of range" panic.
+- `EnsureCapacity(int)`: ensures that the slice have at least the requested capacity.
+- `AppendEmpty() <primitive type>`: adds another slice element with a zero value.
+- `CopyTo(<slice type>)`: copies the slice to another slice of the same type overriding the destination.
+- `MoveTo(<slice type>)`: move the slice to another slice of the same type. The destination is overridden, the 
+  source is set to an empty slice.
+
+### Enum fields
+
+Each protobuf enum field exposed in pdata MUST have a type declared with underlying `int64` type. Name of the type MUST 
+follow the same rules as for struct type names representing protobuf messages:
+
+- If a protobuf enum is defined on the package level, pdata type SHOULD have the same name.
+- If a protobuf enum is defined as part of another message, pdata type SHOULD include the protobuf message name as a 
+  prefix.
+
+Constants defined for the pdata int64 type MUST have the same names and numeric values defined in protobuf. Names 
+of the constants must be translated from ALL_CAPS_SNAKE_CASE to CamelCase.
+
+The pdata enum type SHOULD have `String()` method returning a string value for each constant.
+
+Example of a protobuf enum definition in pdata:
+
+```golang
+type AggregationTemporality int32
+const (
+	AggregationTemporalityUnspecified = AggregationTemporality(otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_UNSPECIFIED)
+	AggregationTemporalityDelta = AggregationTemporality(otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_DELTA)
+	AggregationTemporalityCumulative = AggregationTemporality(otlpmetrics.AggregationTemporality_AGGREGATION_TEMPORALITY_CUMULATIVE)
+)
+```
+
+### Flags
+
+Flags fields are typically defined in protobuf as `uint32` fields representing 32 distinct boolean flags. The bit 
+fields of the flags are defined as enum definitions. The exposed protobuf flags fields MUST be defined as new types 
+with `uint32` underlying type and called according to the same rules as struct names representing protobuf messages 
+and other enum types. Each pdata flags type MUST have an empty variable of the fields type with `Default` prefix. 
+Each flag MUST have a getter method returning a boolean value and a setter method with `With` prefix returning a 
```

Comment:
Not necessary boolean, because we may have non boolean fields in the flags (a 3 state for example).

Reply:
corrected